[Pages] Update `local-development.md` (#8241)
cloudflare/cloudflare-docs: content/pages/platform/functions/local-development.md

$ npx wrangler pages dev <DIRECTORY-OF-ASSETS>
```

This will then start serving your Pages project. You can press `b` to open the browser on your local site.

If you are using a framework, you can pass through the framework-provided dev command (such as, `npm run dev`) to run development. This enables you to benefit from the framework hot-reloading and any special build process around it. To run a framework-provided dev command:

```sh
$ npx wrangler pages dev -- <COMMAND>
```

This will start a local Wrangler development server that serves your Pages project. It will also spawn a new process, running the provided `<COMMAND>`. This provided command is expected to start its own local server, which serves static assets. Wrangler will proxy requests for assets to this underlying server. You can press `b` to open the browser on your local site, (available, by default, on [http://localhost:8788](http://localhost:8788)).

{{<Aside type="note">}}

> `npx wrangler pages dev -- <COMMAND>` does not have support for Pages specific 'metafiles', such as `_headers`, `_redirects` or `_routes.json`. When running in this "proxy" mode, the local development server will ignore any of these files and the respective rules they declare. If your project uses these metafiles, please use the `npx wrangler pages dev <DIRECTORY-OF-ASSETS>` command instead.

Reviewer: ```suggestion
`npx wrangler pages dev -- <COMMAND>` does not have support for Pages specific metafiles, such as `_headers`, `_redirects` or `_routes.json`. When running in this proxy mode, the local development server will ignore any of these files and the respective rules they declare. If your project uses these metafiles, use the `npx wrangler pages dev <DIRECTORY-OF-ASSETS>` command instead.
```

Reviewer: Removing quotation marks around words as style guide asks that we do not give "" answers but rather direct explanations. If proxy isn't the accurate way to describe the mode, just writing 'in this mode' will suffice. Other than this point, PR lgtm!

$ npx wrangler pages dev -- <COMMAND>
```

> This will start a local dev server that serves your Pages project. It will also spawn a new process which will run the framework-provided dev `<COMMAND>`. The dev command will start a local server of its own, which wrangler's dev server will proxy to for serving all the static assets. You can press `B` to open the browser on your local site, or go to http://localhost:8787.

Reviewer 2: ```suggestion
This will start a local Wrangler development server that serves your Pages project. It will also spawn a new process, running the provided `<COMMAND>`. This provided command is expected to start its own server which serves static assets. Wrangler will proxy requests for assets to this underlying server. You can press `b` to open the browser on your local site, (available, by default, on [http://localhost:8788](http://localhost:8788)).
```

{{<Aside type="note">}}

> `npx wrangler pages dev -- <COMMAND>` does not have support for Pages specific metafiles, such as `_headers`, `_redirects` or `_routes.json`. When running in this "proxy" mode, the local development server will ignore any of these files and the respective rules they declare. If you need to include such metafiles in your local development, please use the `npx wrangler pages dev <DIRECTORY-OF-ASSETS>` command instead.

Reviewer 2: ```suggestion
`npx wrangler pages dev -- <COMMAND>` does not have support for Pages specific 'metafiles', such as `_headers`, `_redirects` or `_routes.json`. When running in this "proxy" mode, the local development server will ignore any of these files and the respective rules they declare. If your project uses these metafiles, please use the `npx wrangler pages dev <DIRECTORY-OF-ASSETS>` command instead.
```
